github/attestations.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2024 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"encoding/json"
+)
+
+type Attestation struct {
+	// The attestation's Sigstore Bundle.
+	// Refer to the sigstore bundle specification for more info:
+	// https://github.com/sigstore/protobuf-specs/blob/main/protos/sigstore_bundle.proto
+	Bundle       *json.RawMessage `json:"bundle"`
+	RepositoryID *int             `json:"repository_id"`
+}
+
+type AttestationsResponse struct {
```

Comment:
Is there any reason why we wouldn't want to un-export this struct and return `[]*Attestation` as the first value from the newly-added methods?

I'm fine either way, it just seems like returning a slice might be more useful to end-users of this client library.

Reply:
I had considered it, however its a brand new endpoint and GitHub saw fit to return it as a JSON object instead of an array so it felt like unwrapping it might result in unnecessary breaking changes if GitHub starts including more data in the response.

Reply:
OK, that makes total sense. Let's keep it then.